### ISSUE TITLE
Is it possible to read/write the virtual ram in Tic80?

### ISSUE THREAD

**Issue by another-m00** [nesbox/TIC-80#1866]
Hi, I was wondering if it is possible for an external program to read or write the cartridge ram (not the rom) in Tic80.

I'm asking this because I had an idea to use some of the reserved future use ram space to communicate with an external program that is specifically made for this, like a plugin.

If this is possible, libretro wouldn't need to use pram for communication, and other plugins could be created for example for a file accessing tool.

This actually fits into the fantasy computer perspective too, since all atari, commodore, sinclair, etc computers had an expansion slot which wrote to a custom memory adress.

**Comment by nesbox** [comment]
Hi,
At the moment it is possible only with a `pmem()`, I do not see any other variants, also, we have about 12K reserved [RAM ](https://github.com/nesbox/TIC-80/wiki/ram) for the future, and we could give part of it for such purposes.

**Comment by joshgoebel** [comment]
You can write to the cartridge with `sync`, no?

**Comment by nesbox** [comment]
Yes, in theory you could even write your data to the sprites area for example and then use `sync()` to copy that data to the cart :)

**Comment by NPO-197** [comment]
I would also be interested in having a few bytes of reserved memory dedicated for IO with external programs. Or at least some method of receiving *input* from an external program other then on startup. It would be fun to be able to use the TIC-80 as a programmable plugin for other programs! Or to be able to connect multiple instances of the TIC-80 for multiplayer games.

I will have too look into using `sync()` and `pmem()`, they might be enough for some of the things I'd like to try out for now, awesome work by the way!
